sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDs2to1Transformer.scala

```diff
@@ -0,0 +1,98 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.action.spark.customlogic
+
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.workflow.dataobject.DataObject
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+
+import scala.reflect.runtime.universe.{Type, TypeTag, typeOf}
+
+/**
+ * Interface to define a custom Spark-Dataset transformation (1:1)
+ * When you implement this interface, you need to provide two case classes: One for your input Dataset
+ * and one for your output Dataset.
+ */
+trait CustomDs2to1Transformer[In1 <: Product, In2 <: Product, Out <: Product] extends Serializable {
+
+  /**
+   * Function to be implemented to define the transformation between an input and output DataFrame (1:1)
+   *
+   * @param session      Spark Session
+   * @param options      Options specified in the configuration for this transformation
+   * @param inputDS      Input Dataset
+   * @param dataObjectId Id of DataObject of SubFeed
+   * @return Transformed DataFrame
+   */
+
+  def transform(session: SparkSession, options: Map[String, String], ds1: Dataset[In1], ds2: Dataset[In2]): Dataset[Out]
+
+  def getTypeTag: Type = throw new IllegalArgumentException("When using option \"parameterResolution\" -> \"dataObjectId\"," +
+    " you need to define method getTypeTag next to your transform method as follows: override def getTypeTag: universe.Type = typeOf[this.type]")
+
+  private[smartdatalake] def transformBasedOnDataObjectOrder(session: SparkSession, options: Map[String, String], inputDos: Seq[DataObject], dfs: Map[String, DataFrame])
+                                                            (implicit typeTag1: TypeTag[In1], typeTag2: TypeTag[In2]): DataFrame = {
+    val inputDS1Encoder = org.apache.spark.sql.Encoders.product[In1]
+    val inputDS2Encoder = org.apache.spark.sql.Encoders.product[In2]
+
+    val inputDo1 = inputDos.head
+    val inputDo2 = inputDos(1)
+    val inputDf1 = dfs(inputDo1.id.id)
+    val inputDf2 = dfs(inputDo2.id.id)
+    transform(session, options, inputDf1.as(inputDS1Encoder), inputDf2.as(inputDS2Encoder)).toDF
+  }
+
+  private[smartdatalake] def transformBasedOnDataObjectId(session: SparkSession, options: Map[String, String], dfs: Map[String, DataFrame])
+                                                         (implicit typeTag1: TypeTag[In1], typeTag2: TypeTag[In2]): DataFrame = {
+    val inputDS1Encoder = org.apache.spark.sql.Encoders.product[In1]
+    val inputDS2Encoder = org.apache.spark.sql.Encoders.product[In2]
+
+    val typeTagSubclass = getTypeTag
```

Comment:
Tricky to get the Type of the subclass, yes. But you can use getClass which refers to the subclass, and convert it to a type:
```
    val mirror = scala.reflect.runtime.currentMirror
    val typeTagSubclass = mirror.classSymbol(this.getClass).toType
```